### PR TITLE
Move methods

### PR DESCRIPTION
**Pull request by regis-leray**
@melgenek I added all your changes in this PR.
Please don't hesitate if you have some feedback.

### CODE REVIEW

**Comment by melgenek** [comment]
Hi @regis-leray. Since the https://github.com/zio/zio-s3/pull/156 we started using `.delimiter` and `.startAfter` methods on the `ListObjectsV2Request`. Currently, we need to inject `S3AsyncClient` instead of the `S3.Service`. If you could show how to add these methods to the custom `S3ObjectListing`.

In general I would use the aws-sdk implementation and avoid having the `paginate` as described here https://github.com/zio/zio-s3/pull/156#issuecomment-759594169

File: docs/quickstart/index.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ How to use it ?
 ZIO-S3 is a thin wrapper over the s3 async java client. It exposes the main operations of the s3 java client.
 
 
+```scala
```
Comment (mijicd) [inline docs/quickstart/index.md:21]:
```suggestion
```

File: src/main/scala/zio/s3/package.scala [hunk 2]
```diff
@@ -143,6 +143,44 @@ package object s3 {
...
+       * Read an object by lines
+       *
+       * @param bucketName name of the bucket
+       * @param key: unique key of the object
+       */
+      def streamLines(bucketName: String, key: String): Stream[S3Exception, String] =
+        self
+          .getObject(bucketName, key)
+          .transduce(ZTransducer.utf8Decode)
+          .transduce(ZTransducer.splitLines)
```
Comment (mijicd) [inline src/main/scala/zio/s3/package.scala:156]:
```suggestion
          .transduce(ZTransducer.utf8Decode >>> ZTransducer.splitLines)
```

File: src/main/scala/zio/s3/package.scala [hunk 3]
```diff
@@ -143,6 +143,44 @@ package object s3 {
...
+      /**
+       * List all descendant objects of a bucket
+       * Fetch all objects recursively of all nested directory by traversing all of them
+       *
+       * @param bucketName name of the bucket
+       * @param prefix filter all object identifier which start with this `prefix`
+       */
+      def listAllObjects(bucketName: String, prefix: String): Stream[S3Exception, S3ObjectSummary] =
+        ZStream
+          .fromEffect(self.listObjects(bucketName, prefix, 1000))
```
Comment (mijicd) [inline src/main/scala/zio/s3/package.scala:167]:
```suggestion
          .fromEffect(self.listObjects(bucketName, prefix))
```

**Comment by regis-leray** [comment]
@melgenek i supported delimiter / startWith expose as parameters and also support first version of the behavior inside the s3 Mock implementation

**Pull request merged**